syntax = "proto3";

package tweety;

import "models.proto";

service MessageService {
  rpc PostMessage (PostMessageRequest) returns (PostMessageResponse);
  rpc GetMessages (GetMessagesRequest) returns (GetMessagesResponse);
  rpc RepostMessage (RepostMessageRequest) returns (RepostMessageResponse);
}

message GetMessagesRequest {
  string username = 1; // Optional: Can be used to filter messages
}

message GetMessagesResponse {
  repeated Message messages = 1;
}

message PostMessageRequest {
  string username = 1;
  string content = 2;
}

message PostMessageResponse {
  bool success = 1;
  string message = 2;
}

message RepostMessageRequest {
  string username = 1; // User who is reposting
  int32 original_message_index = 2; // Index of the original message in the messages array
}

message RepostMessageResponse {
  bool success = 1;
  string message = 2;
}


